enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/UnsecuredAPIAuthenticator.java

```diff
@@ -79,7 +86,29 @@ public AuthenticationContext authenticate(RequestContext requestContext) throws
                 throw new APISecurityException(APIConstants.StatusCodes.SERVICE_UNAVAILABLE.getCode(),
                         GeneralErrorCodeConstants.API_BLOCKED_CODE, GeneralErrorCodeConstants.API_BLOCKED_MESSAGE);
             }
-            return FilterUtils.generateAuthenticationContextForUnsecured(requestContext);
+
+            // Generate or get backend JWT
+            JWTConfigurationDto backendJwtConfig = ConfigHolder.getInstance().getConfig().
+                    getJwtConfigurationDto();
+            String endUserToken = null;
+
+            if (backendJwtConfig.isEnabled()) {
```

Comment:
Shall we move this to the class constructor like this? https://github.com/wso2/product-microgateway/blob/e190b0b250a9a5dd26ff73f9b2612e562cf73860/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L84